packages/wallet-specs/src/protocols/depositing/protocol.ts

```diff
@@ -110,32 +86,34 @@ type Services = {
 
 type Guards = {
   safeToDeposit: (context: Init) => boolean;
-  funded: (context: Init) => boolean;
 };
 type Options = { services: Services; guards: Guards };
 export const machine: MachineFactory<Init, any> = (store: IStore, context: Init) => {
   const subscribeDepositEvent = (context: Init) => callback => {
-    return store.on('DEPOSITED', event => {
+    return store.on('DEPOSITED', (event: ChainEvent) => {
       if (event.type === 'DEPOSITED' && event.channelId === context.channelId) {
-        callback('DEPOSITED');
+        if (bigNumberify(event.total).gte(context.fundedAt)) {
```

Comment:
I know we were talking before about making our protocols less reliant on events from the blockchain and instead relying on the state. We could this here by just dispatching one event (`CHAIN_UPDATED`)  but that makes the state machine more complex. Any thoughts?

Reply:
We could definitely do that. Even if we do, it could be nice for this callback to send `'FUNDED` or `'DEPOSITED` to the machine.